### ISSUE TITLE
Add a Number Box Widget

### ISSUE THREAD

**Issue by slimyjimmy** [bdlukaa/fluent_ui#560]
It would be great if there was a dedicated widget for numerical inputs.

See this Microsoft article as reference: https://learn.microsoft.com/en-us/windows/apps/design/controls/number-box

**Comment by WinXaito** [comment]
Yup, I already tried here: https://github.com/bdlukaa/fluent_ui/pull/212 

But I had some problem with focus on this and I currently don't have time to work on it. (I will need that one time for sure, but not currently).

If anyone would work on that, you can take a look on this PR if that can help you ;)
